### PR TITLE
Use key vault task in CI yaml

### PR DESCRIPTION
**Pull request by damonbarry**
Currently our integration tests expect a certificate to be installed on the test machine. The certificate gives us access to a key vault containing the secrets we need for the tests. The certificate itself is also in the key vault, so we have to give the pipeline access to get the certificate. All this was done early in the project, before the key vault task existed in Azure Pipelines.

Using the Azure Pipelines Key Vault task allows us to remove key vault credentials from the pipeline. It also simplifies the build machine setup: we no longer need to install the certificate, and on Linux we don't need to install things like Python, the `az` CLI, and PowerShell that our scripts require to get/install the certificate. Instead we use the task to list exactly the secrets we need, and they are available to the pipeline. For script tasks, we map the secrets into the environment so they don't get passed as arguments and potentially exposed.

Other changes:
- Use the .NET Core installer task to pin the version of .NET Core for builds
- Remove "Install dependencies" and "install test dependencies" tasks since they're no longer needed. For Linux I had to add a script task to install libsnappy, which installPrereqs.sh was doing. It's needed for the rocksdb integration tests. I also had to add a task to install the protocol head certificate, which several integration tests need. This could arguably be checked in as a PS1 script rather than inline script in the YAML file, but this works for now.
- Remove tasks that copy and publish artifacts for the end-to-end tests (IotEdgeQuickstart and friends). No other pipelines curently consume it right now (they rely on the Build Images pipeline). We can add these back if/when they're needed.
- Move the Windows job onto a hosted agent
- Update SecretsHelper to look in the environment first for the needed secrets. When the integration tests are run in the CI pipeline, the KeyVaultHelper is never invoked. Elsewhere, things continue to work like they always have.
- Our build scripts expect dotnet.exe to live in a certain path, but the .NET Core Installer task puts it somewhere funny, and then prepends that location to the PATH. So I updated the build/test scripts to (1) look for an environment variable, then (2) look on the PATH, then (3) look in some default locations.

### CODE REVIEW

File: builds/ci/dotnet.yaml [hunk 1]
```diff
@@ -5,110 +5,108 @@ trigger:
...
-        displayName: 'Build'
-      - script: sudo -E bash -c './scripts/linux/runTests.sh "--filter $TEST_FILTER" "$BUILD_CONFIGURATION"'
-        name: test
+      - task: AzureKeyVault@1
+        displayName: Get secrets
+        inputs:
+          azureSubscription: $(azure.subscription)
+          keyVaultName: $(kv.name)
+          secretsFilter: >-
+            DummySecret1,
```
Comment (philipktlin) [inline builds/ci/dotnet.yaml]: what is DummySecret1 used for?

Comment (damonbarry) [reply builds/ci/dotnet.yaml]: To test SecretsHelper

File: builds/ci/dotnet.yaml [hunk 2]
```diff
@@ -5,110 +5,108 @@ trigger:
...
+          IotHubConnStr2: $(IotHubConnStr2)
+          IotHubMqttHeadCert: $(IotHubMqttHeadCert)
       - task: PublishTestResults@1
-        displayName: 'Publish test results'
+        displayName: Publish test results
         inputs:
           testRunner: VSTest
           testResultsFiles: '**/TestResults/result.trx'
         condition: succeededOrFailed()
-      - task: PublishBuildArtifacts@1
```
Comment (philipktlin) [inline builds/ci/dotnet.yaml]: Nice clean up!

File: builds/ci/dotnet.yaml [hunk 3]
```diff
@@ -5,110 +5,108 @@ trigger:
...
+            IotHubConnStr2,
+            IotHubMqttHeadCert
+      - task: DotNetCoreInstaller@0
+        displayName: Install .NET Core
         inputs:
-          azureSubscription: "$(azure.subscription)"
-          ScriptPath: 'scripts/windows/setup/Install-VaultCertificate.ps1'
-          ScriptArguments: '-VaultName $(kv.name) -CertificateName IoTEdgeTestCert'
-          azurePowerShellVersion: LatestVersion
-      - powershell: scripts/windows/setup/Install-Prerequisites.ps1 -DotnetSdkUrl $(NetCorePackageUri)
```
Comment (philipktlin) [inline builds/ci/dotnet.yaml]: No longer need this file "Install-Prerequisites.ps1"? remove it?

Comment (damonbarry) [reply builds/ci/dotnet.yaml]: Well, those scripts are still being used in lots of places...this PR is _only_ the CI pipeline. When we clean it up everywhere, then we can remove.

**Comment by myagley** [comment]
There is so much good stuff in here. Thanks! Super excited to get this on the hosted pool.

**Pull request merged**